relayer/relays/beacon/header/header.go

```diff
@@ -552,12 +552,21 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
 		return beaconState, fmt.Errorf("GetLastFinalizedStateIndex error: %w", err)
 	}
 	startIndex := uint64(lastIndex)
+
 	endIndex := startIndex + 1
+	iterations := uint64(0)
 
 	syncCommitteePeriod := h.protocol.Settings.SlotsInEpoch * h.protocol.Settings.EpochsPerSyncCommitteePeriod
 	totalStates := h.protocol.Settings.EpochsPerSyncCommitteePeriod * h.protocol.HeaderRedundancy // Total size of the circular buffer,
 	// https://github.com/paritytech/polkadot-sdk/blob/master/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L75
 	for index := startIndex; index != endIndex; index = (index - 1 + totalStates) % totalStates {
```

Comment:
I'd suggest to simplify the loop pattern to end at the startIndex.

```
for {
    ....
    index = (index - 1 + totalStates) % totalStates
    if index == startIndex {
	    log.Debug("loop finished, max iterations reached")
	    break
    }
}
...
```

Reply:
That's a good point. 😅 Let me know what you think of https://github.com/Snowfork/snowbridge/pull/1304/commits/ba114cf1b9e28f0074f35d9be860eac8b6fe6769.